library/geocore/src/main/java/it/geosolutions/imageio/compression/Compressor.java

```diff
@@ -0,0 +1,61 @@
+/*
+ *    ImageI/O-Ext - OpenSource Java Image translation Library
+ *    https://www.geosolutionsgroup.com/
+ *    https://github.com/geosolutions-it/imageio-ext
+ *    (C) 2022, GeoSolutions
+ *    All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *     * Redistributions of source code must retain the above copyright
+ *       notice, this list of conditions and the following disclaimer.
+ *     * Redistributions in binary form must reproduce the above copyright
+ *       notice, this list of conditions and the following disclaimer in the
+ *       documentation and/or other materials provided with the distribution.
+ *     * Neither the name of GeoSolutions nor the
+ *       names of its contributors may be used to endorse or promote products
+ *       derived from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY GeoSolutions ``AS IS'' AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL GeoSolutions BE LIABLE FOR ANY
+ * DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package it.geosolutions.imageio.compression;
+
+/**
+ * Interface for byte array Compressor.
+ *
+ * Compressor should be first initialized using setInput with a source data byte array.
+ * Afterwards, the compress method should be called to compress source data and result stored
+ * on destData.
+ */
+public interface Compressor {
+
+    void setInput(byte[] srcData);
+
+    /**
+     * indicates that compression should end with the current contents of the input buffer.
+     * Some implementations may need it.
+     */
+    void finish();
+
+    /**
+     * compress the input source data section selected by srcOffset and srcLength and
+     * store the on destData, given destOffset and destLength.
+     */
+    int compress(byte[] destData, int srcOffset, int srcLength, int destOffset, int destlength);
```

Comment:
Thinking out loud, why provide the srcData as a property, but the destdata as a parameter?
Would it make sense to have both as params?

Reply:
the libdeflate implementation may need to call finish before doing the compression and it needed to get the source data first.